tests/test_qt_plotting.py

```diff
@@ -21,18 +21,23 @@ class QMainWindow(object):
 
 
 class TstWindow(MainWindow):
-    def __init__(self, parent=None, show=True):
+    def __init__(self, parent=None, show=True, off_screen=True):
         MainWindow.__init__(self, parent)
 
         self.frame = QFrame()
         vlayout = QVBoxLayout()
-        self.vtk_widget = QtInteractor(self.frame)
+        self.vtk_widget = QtInteractor(
+            parent=self.frame,
+            off_screen=off_screen
+        )
         vlayout.addWidget(self.vtk_widget.interactor)
 
         self.frame.setLayout(vlayout)
         self.setCentralWidget(self.frame)
 
-        mainMenu = self.menuBar()
+        mainMenu = QMenuBar(parent=self)
+        mainMenu.setNativeMenuBar(False)
```

Comment:
I added a test for that based on `isNativeMenuBar()`, thank you so much for reporting that.

Reply:
This indeed covers that exact problem. However if someone manages to fix the underlying problem and reenables native menus, this test will needlessly fail. Using the same code to create the menu in production and tests would catch the underlying error, without being specific to the native menu workaround.

Reply:
I'm not sure to understand what you mean. The same code is used in testing and production since the menu bar is created in `BackgroundPlotter` and this is exactly the target of the test.

Also, I considered `setNativeMenuBar(False)` as the fix for the underlying issue for the reasons I mentioned in https://github.com/pyvista/pyvista/pull/619#issuecomment-590879215
But if someone `setNativeMenuBar(True)` the rest of the test (not only `assert isNativeMenuBar(False)`) is expected to fail because lifetime and ownership is tested.

Sorry if I miss your point, feel free to push in here if you had something different in mind.

Reply:
If lifetime and ownership tests are impossible to work with native menu bars, then this is likely a non-issue indeed - I wasn't aware of that.

My point is that the lines (well just 2, so maybe a bit pedantic, sorry...) we are discussing here are the same as in `__init__` of `BackgroundPlotter`, assumingly to mock that behaviour (thought that assumption may again be wrong): https://github.com/pyvista/pyvista/blob/a8b96dcc2b71e94633a677c5d38d3f22281ba3bd/pyvista/plotting/qt_plotting.py#L677 So to correctly mock it even if that code is changed, I'd extract that code into a helper function (e.g. `_create_menu_bar()`) that's used both here and on `BackgroundPlotter`.

Reply:
You're talking about `TstWindow`! I missed that completely.

+1 for `_create_menu_bar()`

Reply:
> If lifetime and ownership tests are impossible to work with native menu bars, then this is likely a non-issue indeed - I wasn't aware of that.

Especially on `macOS`: https://github.com/pyvista/pyvista/pull/619#issuecomment-590056370

Reply:
It took longer than necessary but I finally added `_create_menu_bar()`.